src/util/streamable_generator.h

```diff
@@ -44,16 +48,20 @@ class suspend_sometimes {
  */
 template <size_t MIN_BUFFER_SIZE>
 class stream_generator_promise {
-  ad_utility::streams::ostringstream _value;
+  std::string _value;
+  io::filtering_ostream _filterStream;
   std::exception_ptr _exception;
 
  public:
-  stream_generator_promise() = default;
+  stream_generator_promise() { setContentEncoding(CompressionMethod::NONE); }
 
   basic_stream_generator<MIN_BUFFER_SIZE> get_return_object() noexcept;
 
   constexpr std::suspend_always initial_suspend() const noexcept { return {}; }
-  constexpr std::suspend_always final_suspend() const noexcept { return {}; }
+  std::suspend_always final_suspend() noexcept {
+    _filterStream.reset();
```

Comment:
Comment, that reset also logically flushes, s.t. the remaining contents now reside in the `_value`.